site/src/docs/foundation/guidelines/server-side-rendering.mdx

```diff
@@ -0,0 +1,171 @@
+or pre-rendering), or during an HTTP request. The alternative to SSR is client-side rendering (CSR), where most of the HTML
+content is composed and rendered in the browser using JavaScript.
+
+## How does HDS support server-side rendering?
+
+For HDS components to work with server-side rendering, in server you need to inject the critical CSS styles
+of the HDS components that are being used into the initially rendered HTML. For hds-core, you must include the styles from
+provided CSS files yourself. For hds-react, HDS provides multiple options, that we will cover next.
+
+<Notification label="Critical CSS" className="siteNotification">
+Critical CSS is the CSS applied to above-the-fold elements. It is the CSS of the immediately visible content in the browser
+viewport when the user opens your website. Critical CSS does not usually have the styles of the elements that are in the
+scrollable content outside of the browser viewport. There is one exception though, and that is if you have anchor links on the page.
+When user opens website from an url that includes an anchor link, the browser automatically scrolls the page so that the link
+will be visible.
+</Notification>
+
+### Option 1: getCriticalHdsRules tool to get critical styles (recommended)
+
+HDS React components use CSS-in-JS in a way where the styles are by default injected into the head tag in the browser.
+If you use server-side rendering and do not include the critical styles on the server, this leads to flashes of unstyled
+content when the user lands on the page.
+
+To include critical styles in server, HDS exposes a tool for extracting the used critical styles of HDS components.
+This should come handy in multiple ways:
+
+- Automatically extracts styles based on used HTML.
+- If you add more HDS components later, you do not need to remember to add their styles separately.
+- If you happen to remove HDS components, you do not need to remember to remove their respective styles.
+
+Let's go through a simple example of the usage of the tool:
+
+```js
+import { getCriticalHdsRules, hdsStyles } from "hds-react";
+
+const criticalHdsRules = await getCriticalHdsRules(bodyHTML, hdsStyles);
+
+const finalHTML =
+<html>
+  <head>
+    ...
+    <style data-used-styles dangerouslySetInnerHTML={{ __html: criticalHdsRules }} />
+  </head>
+  <body>
+    ...
+  </body>
+<html>
+```
+
+First, we import the tool that is named `getCriticalHdsRules`. To be able to call it, we need two things:
+
+1. The initially to be rendered HTML body code as a string.
+2. hdsStyles - a variable holding all the styles of the HDS react components as a string.
+
+Calling it returns a string containing the critical CSS styles. The rest is easy, we set those styles into
+a style tag, that we will then inject into the finally rendered HTML document. It can be wise to cache the result of
+`getCriticalHdsRules` based on the function parameters, to improve performance.
```

Comment:
This might nit picking and completely up to preference but I feel like this reads smoother: "--based on the function parameters in order to improve performance." But go ahead and use whatever feels more natural to you.

Reply:
- [ ] based on the function parameters, to improve performance. -> based on the function parameters in order to improve performance.